flytekit/extras/tensorflow/model.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+from typing import Type
+
+import tensorflow as tf
+
+from flytekit.core.context_manager import FlyteContext
+from flytekit.core.type_engine import TypeEngine, TypeTransformer, TypeTransformerFailedError
+from flytekit.models.core import types as _core_types
+from flytekit.models.literals import Blob, BlobMetadata, Literal, Scalar
+from flytekit.models.types import LiteralType
+
+
+class TensorflowModelTransformer(TypeTransformer[tf.keras.Model]):
```

Comment:
```suggestion
class TensorflowModelTransformer(TypeTransformer[tf.Module]):
```

Please check the comments we've added.

Reply:
I tried using the `tf.Module` but the `tf.keras.models.save_model` method fails when saving the `Layers` (in the unit tests).
I don't think we can save the whole layer in tensorflow, we can only save its weights

How should I go about it ?
1. keep the type as `tf.keras.Model`
2. or use `tf.Module` and add separate check for `Layers` and save them separately

Reply:
@techytushar, you're right. How about then you create two files, one each for model and layers? You can name as `model.py` and `layer.py`, respectively. The model transformer also needs to subclass `tf.keras.Sequential` along with `tf.keras.Model`. Please fine the layer serialization [here](https://www.tensorflow.org/api_docs/python/tf/keras/layers/serialize).

Reply:
For layer one, set the literal to `generic`, e.g., https://github.com/flyteorg/flytekit/blob/6530601c2538a5d804127a97f63291730b1ba1d8/plugins/flytekit-dolt/flytekitplugins/dolt/schema.py#L76.

Reply:
The `Sequential` [class](https://www.tensorflow.org/api_docs/python/tf/keras/Sequential) already inherits from the `Model` class, so is the subclassing needed ? If it is a `Sequential` type then it automatically use the `ModelTransformer`

Reply:
Oh, then not needed.